include/roboteam_world/kalman/kalmanFilter.h

```diff
@@ -28,14 +26,16 @@ namespace rtt {
 
         void kalmanUpdate();
 
-        void newFrame(const roboteam_msgs::DetectionFrame& msg);
+        void newFrame(const roboteam_proto::SSL_DetectionFrame& msg);
 
-        roboteam_msgs::World getWorld();
+        roboteam_proto::World getWorld();
+        std::string getWorldMsg();
 
         kalmanThem theirBots[BOTCOUNT];
         kalmanUs ourBots[BOTCOUNT];
         kalmanBall ball;
 
+        std::mutex filterMutex;
```

Comment:
Why did we need a filterMutex? When you tuned up the performance, was it trying to update itself too often?

Reply:
Yes